src/main/java/org/spongepowered/common/data/processor/data/entity/GravityDataProcessor.java

```diff
@@ -0,0 +1,84 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.data.entity;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.nbt.NBTTagCompound;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableGravityData;
+import org.spongepowered.api.data.manipulator.mutable.entity.GravityData;
+import org.spongepowered.api.data.value.ValueContainer;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeGravityData;
+import org.spongepowered.common.data.processor.common.AbstractEntitySingleDataProcessor;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+import org.spongepowered.common.data.value.mutable.SpongeValue;
+
+import java.util.Optional;
+
+public class GravityDataProcessor extends AbstractEntitySingleDataProcessor<Entity, Boolean, Value<Boolean>, GravityData, ImmutableGravityData> {
+
+    public GravityDataProcessor() {
+        super(Entity.class, Keys.ENTITY_HAS_GRAVITY);
+    }
+
+    @Override
+    protected Value<Boolean> constructValue(Boolean actualValue) {
+        return new SpongeValue<>(Keys.ENTITY_HAS_GRAVITY, actualValue);
+    }
+
+    @Override
+    protected GravityData createManipulator() {
+        return new SpongeGravityData();
+    }
+
+    @Override
+    protected boolean set(Entity entity, Boolean value) {
+        NBTTagCompound compound = new NBTTagCompound();
+        entity.readFromNBT(compound);
+        compound.setBoolean("NoGravity", !value);
+        entity.writeToNBT(compound);
+        return true;
+    }
+
+    @Override
+    protected Optional<Boolean> getVal(Entity entity) {
+        NBTTagCompound compound = new NBTTagCompound();
+        entity.readFromNBT(compound);
```

Comment:
There are methods for working with the "has no gravity" property.
net/minecraft/entity/Entity.hasNoGravity
net/minecraft/entity/Entity.setNoGravity